File: app/src/main/java/com/nwise/mvptemplate/domain/executer/NetworkJobExecutor.java
package com.nwise.mvptemplate.domain.executer;


import io.reactivex.Scheduler;
import io.reactivex.schedulers.Schedulers;


public class NetworkJobExecutor implements UseCaseExecutor {

    @Override
    public Scheduler getScheduler() {
        return Schedulers.io();
    }
}

File: app/src/main/java/com/nwise/mvptemplate/domain/executer/NetworkJobExecutor.kt
package com.nwise.mvptemplate.domain.executer

import io.reactivex.Scheduler
import io.reactivex.schedulers.Schedulers

/**
 * Created by Sepideh Vatankhah on 01.08.2019.
 * sun.vatankhah@gmail.com
 * https://github.com/sepidevatankhah
 */
class NetworkJobExecutor : UseCaseExecutor {

    override val scheduler: Scheduler
        get() = Schedulers.io()
}
